roles/servicetelemetry/tasks/component_clouds.yml

```diff
@@ -19,20 +19,51 @@
     - this_cloud.metrics.collectors is iterable
 
 - name: Events Smart Gateway deployment
+  when: servicetelemetry_vars.backends.events.elasticsearch.enabled
   block:
     - name: Lookup ElasticSearch BasicAuth
       k8s_info:
         api_version: v1
         kind: Secret
         namespace: '{{ ansible_operator_meta.namespace }}'
-        name: 'elasticsearch-es-elastic-user'
+        name: '{{ servicetelemetry_vars.backends.events.elasticsearch.extUserSecretName }}'
       register: elasticsearch_es_elastic_user
 
-    - name: Filter out ElasticSearch password for BasicAuth
+    - name: Set ElasticSearch user for BasicAuth
       set_fact:
-        elastic_pass: "{{ elasticsearch_es_elastic_user | json_query('resources[0].data.elastic') | b64decode }}"
+        elastic_user: "{{ elasticsearch_es_elastic_user.resources[0].data | dict2items | map(attribute='key') | list | first }}"
+
+    - name: Set ElasticSearch password for BasicAuth
+      set_fact:
+        elastic_pass: "{{ elasticsearch_es_elastic_user.resources[0].data[elastic_user] | b64decode }}"
       no_log: true
 
+    - name: Get the ElasticSearch TLS materials secret
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        namespace: '{{ ansible_operator_meta.namespace }}'
+        name: '{{ servicetelemetry_vars.backends.events.elasticsearch.extTlsSecretName }}'
+      register: es_certs
+
+    - when: es_certs[0].data[user.key] is not defined or es_certs[0].data[user.crt] is not defined
+      block:
+        - name: Load dummy certs
+          include_vars:
+            file: dummy_user_certs.yml
```

Comment:
Do you want to maybe add a check here to see if only one of these is set? If that's the case I would think this include_vars overrides the user-passed data, which I think makes sense because you should probably be passing in both values not just one.

Reply:
The 'when' clause is already such that both must be supplied, otherwise both will be replaced by the dummies. I think this is what you're suggesting, but I might have misunderstood.